src/renderers/dom/shared/ReactDOMComponentTree.js

```diff
@@ -30,12 +30,9 @@ var internalEventHandlersKey = '__reactEventHandlers$' + randomKey;
  * Check if a given node should be cached.
  */
 function shouldPrecacheNode(node, nodeID) {
-  return (node.nodeType === 1 &&
-          node.getAttribute(ATTR_NAME) === ('' + nodeID)) ||
-         (node.nodeType === 8 &&
-          node.nodeValue === ' react-text: ' + nodeID + ' ') ||
-         (node.nodeType === 8 &&
-          node.nodeValue === ' react-empty: ' + nodeID + ' ');
+  return (node.nodeType === 1 && node.getAttribute(ATTR_NAME) === '' + nodeID) ||
+    (node.nodeType === 8 && node.nodeValue === ' react-text: ' + nodeID + ' ') ||
+    (node.nodeType === 8 && node.nodeValue === ' react-empty: ' + nodeID + ' ');
```

Comment:
@jlongster I really like the decision we made to add parenthesis around && even though they are not necessary. People tend to put them anyway.